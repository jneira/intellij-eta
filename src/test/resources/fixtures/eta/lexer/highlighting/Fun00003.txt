ITinline_prag ('{-#')
ITline_prag (' LANGUAGE GADTs ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE KindSignatures ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE RankNTypes ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE ScopedTypeVariables ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE NPlusKPatterns ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE BangPatterns ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE ViewPatterns ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE PatternGuards ')
ITline_prag ('#-}')
WHITE_SPACE ('\n\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00003')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Show')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Info')
WHITE_SPACE (' ')
ITvarid ('a')
ITcparen (')')
WHITE_SPACE ('\n  ')
ITwhere ('where')
WHITE_SPACE ('\n    ')
ITvarid ('show')
WHITE_SPACE (' ')
ITvarid ('i')
ITat ('@')
IToparen ('(')
ITconid ('Info')
WHITE_SPACE (' ')
ITocurly ('{')
ITccurly ('}')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('show')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('infoType')
WHITE_SPACE (' ')
ITvarid ('i')
ITcparen (')')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('szStr')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('srcStr')
WHITE_SPACE ('\n      ')
ITwhere ('where')
WHITE_SPACE ('\n        ')
ITvarid ('szStr')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('show')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('infoSize')
WHITE_SPACE (' ')
ITvarid ('i')
ITcparen (')')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n          ')
ITstring ('"')
ITstring ('AnySize')
ITstring ('"')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('"')
ITstring ('"')
WHITE_SPACE ('\n          ')
ITvarid ('str')
WHITE_SPACE ('       ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('"')
ITstring (' | ')
ITstring ('"')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('str')
WHITE_SPACE ('\n\n        ')
ITvarid ('srcStr')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('infoSource')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n          ')
ITstring ('"')
ITstring ('"')
WHITE_SPACE ('  ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('"')
ITstring ('"')
WHITE_SPACE ('\n          ')
ITvarid ('src')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('"')
ITstring (' | ')
ITstring ('"')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('src')
WHITE_SPACE ('\n\n')
ITdata ('data')
WHITE_SPACE (' ')
ITconid ('Set')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('cxt')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITvarsym ('*')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
IToparen ('(')
ITvarsym ('*')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarsym ('*')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Set')
WHITE_SPACE (' ')
ITobrack ('[')
ITvarid ('a')
ITcbrack (']')
WHITE_SPACE ('\n\n')
ITvarid ('g2')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
ITforall ('forall')
WHITE_SPACE (' ')
ITvarid ('a')
ITvarsym ('.')
WHITE_SPACE (' ')
ITconid ('Eq')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITobrack ('[')
ITvarid ('a')
ITcbrack (']')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Bool')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE ('\n')
ITvarid ('g2')
WHITE_SPACE (' ')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITvbar ('|')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('m')
WHITE_SPACE (' ')
ITlarrow ('<-')
WHITE_SPACE (' ')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('undefined')
WHITE_SPACE ('\n       ')
ITcomma (',')
WHITE_SPACE (' ')
ITconid ('True')
WHITE_SPACE ('\n       ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n\n')
ITvarid ('g3')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
ITforall ('forall')
WHITE_SPACE (' ')
ITvarid ('a')
ITvarsym ('.')
WHITE_SPACE (' ')
ITconid ('Eq')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITobrack ('[')
ITvarid ('a')
ITcbrack (']')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Bool')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITvarsym ('*')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE ('\n')
ITvarid ('g3')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('undefined')
WHITE_SPACE ('\n\n')
ITvarid ('f')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarsym ('\')
IToparen ('(')
ITvarid ('x')
ITdcolon ('::')
ITconid ('Int')
ITcomma (',')
WHITE_SPACE (' ')
ITvarid ('y')
ITdcolon ('::')
ITvarid ('a')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n')
ITvarid ('g')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('x')
ITdcolon ('::')
ITvarid ('a')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n')
ITvarid ('h')
WHITE_SPACE (' ')
IToparen ('(')
IToparen ('(')
ITvarid ('x')
ITcomma (',')
ITvarid ('y')
ITcparen (')')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Int')
ITcomma (',')
ITconid ('Bool')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('y')
ITcomma (',')
ITvarid ('x')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITvarid ('fac')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('n')
ITvarsym ('+')
ITinteger ('1')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('n')
ITvarsym ('+')
ITinteger ('1')
ITcparen (')')
WHITE_SPACE (' ')
ITvarsym ('*')
WHITE_SPACE (' ')
ITvarid ('fac')
WHITE_SPACE (' ')
ITvarid ('n')
WHITE_SPACE ('\n\n')
ITvarid ('foc')
WHITE_SPACE (' ')
IToparen ('(')
ITvarsym ('-')
ITinteger ('1')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('3')
WHITE_SPACE ('\n\n')
ITvarid ('lazyFunc')
ITcomma (',')
WHITE_SPACE (' ')
ITvarid ('strictFunc')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
IToparen ('(')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITvarid ('lazyFunc')
WHITE_SPACE (' ')
ITtilde ('~')
IToparen ('(')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITvarid ('strictFunc')
WHITE_SPACE (' ')
ITvarsym ('!')
ITvarid ('v')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITvarid ('firstZero')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('head')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITinteger ('0')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('True')
