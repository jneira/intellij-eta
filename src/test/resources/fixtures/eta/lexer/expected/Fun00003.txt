ITblockComment ('{-# LANGUAGE GADTs #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE KindSignatures #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE RankNTypes #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE ScopedTypeVariables #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE NPlusKPatterns #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE BangPatterns #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE ViewPatterns #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE PatternGuards #-}')
WHITE_SPACE ('\n\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00003')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITvocurly ('')
WHITE_SPACE ('')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Show')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Info')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n  ')
ITwhere ('where')
WHITE_SPACE ('\n    ')
ITvocurly ('')
WHITE_SPACE ('')
ITvarid ('show')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE ('')
ITat ('@')
WHITE_SPACE ('')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Info')
WHITE_SPACE (' ')
ITocurly ('{')
WHITE_SPACE ('')
ITccurly ('}')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('show')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('infoType')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('szStr')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('srcStr')
WHITE_SPACE ('\n      ')
ITwhere ('where')
WHITE_SPACE ('\n        ')
ITvocurly ('')
WHITE_SPACE ('')
ITvarid ('szStr')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('show')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('infoSize')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n          ')
ITvocurly ('')
WHITE_SPACE ('')
ITstring ('"AnySize"')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('""')
WHITE_SPACE ('\n          ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('str')
WHITE_SPACE ('       ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('" | "')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('str')
WHITE_SPACE ('\n\n        ')
ITvccurly ('')
WHITE_SPACE ('')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('srcStr')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('infoSource')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n          ')
ITvocurly ('')
WHITE_SPACE ('')
ITstring ('""')
WHITE_SPACE ('  ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('""')
WHITE_SPACE ('\n          ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('src')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstring ('" | "')
WHITE_SPACE (' ')
ITvarsym ('++')
WHITE_SPACE (' ')
ITvarid ('src')
WHITE_SPACE ('\n\n')
ITvccurly ('')
WHITE_SPACE ('')
ITvccurly ('')
WHITE_SPACE ('')
ITvccurly ('')
WHITE_SPACE ('')
ITsemi ('')
WHITE_SPACE ('')
ITdata ('data')
WHITE_SPACE (' ')
ITconid ('Set')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('cxt')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITstar ('*')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITstar ('*')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITstar ('*')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Set')
WHITE_SPACE (' ')
ITobrack ('[')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE ('')
ITcbrack (']')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g2')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('forall')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITdot ('.')
WHITE_SPACE (' ')
ITconid ('Eq')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITobrack ('[')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE ('')
ITcbrack (']')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Bool')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g2')
WHITE_SPACE (' ')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITvbar ('|')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('m')
WHITE_SPACE (' ')
ITlarrow ('<-')
WHITE_SPACE (' ')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('undefined')
WHITE_SPACE ('\n       ')
ITcomma (',')
WHITE_SPACE (' ')
ITconid ('True')
WHITE_SPACE ('\n       ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g3')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('forall')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITdot ('.')
WHITE_SPACE (' ')
ITconid ('Eq')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITobrack ('[')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE ('')
ITcbrack (']')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Bool')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITstar ('*')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g3')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('undefined')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('f')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITlam ('\')
WHITE_SPACE ('')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE ('')
ITdcolon ('::')
WHITE_SPACE ('')
ITconid ('Int')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE (' ')
ITvarid ('y')
WHITE_SPACE ('')
ITdcolon ('::')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE ('')
ITdcolon ('::')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('h')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE ('')
ITvarid ('y')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Int')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE ('')
ITconid ('Bool')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('y')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('fac')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('n')
WHITE_SPACE ('')
ITvarsym ('+')
WHITE_SPACE ('')
ITinteger ('1')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('n')
WHITE_SPACE ('')
ITvarsym ('+')
WHITE_SPACE ('')
ITinteger ('1')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITstar ('*')
WHITE_SPACE (' ')
ITvarid ('fac')
WHITE_SPACE (' ')
ITvarid ('n')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('foc')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITminus ('-')
WHITE_SPACE ('')
ITinteger ('1')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('3')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('lazyFunc')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE (' ')
ITvarid ('strictFunc')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('lazyFunc')
WHITE_SPACE (' ')
ITtilde ('~')
WHITE_SPACE ('')
IToparen ('(')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('strictFunc')
WHITE_SPACE (' ')
ITbang ('!')
WHITE_SPACE ('')
ITvarid ('v')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('firstZero')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('head')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITinteger ('0')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('True')